### PR TITLE
LPS-89659 SchedulerEventMessageListenerWrapper bug

### PR DESCRIPTION
**Pull request by dantewang**
@tinatian If a thread which is trying to lock is interrupted, it will still execute _processMessage() and then unlock. Hai found this when trying to add code coverage.

### CODE REVIEW

**Comment by dantewang** [comment]
ci:test:sf

**Comment by dantewang** [comment]
ci:test:relevant

File: portal-kernel/test/unit/com/liferay/portal/kernel/scheduler/messaging/SchedulerEventMessageListenerWrapperTest.java [hunk 1]
```diff
@@ -185,19 +240,19 @@ public void testConcurrentReceiveWithTimeout() throws Exception {
...
-					_blockCountDownLatch.await();
-
-					message.setResponse(message);
-				}
-				catch (InterruptedException ie) {
-				}
-				finally {
-					_lock.unlock();
-				}
+			_lock.lock();
```
Comment (tinatian) [inline portal-kernel/test/unit/com/liferay/portal/kernel/scheduler/messaging/SchedulerEventMessageListenerWrapperTest.java:243]: Hi @yuhai,
Why you made this change?

The reason using tryLock here is to make sure that listener is not invoked concurrently, if it is invoked concurrently, then only one message is process.

Tina.

**Comment by tinatian** [comment]
See https://github.com/shuyangzhou/liferay-portal/pull/7383

File: portal-kernel/test/unit/com/liferay/portal/kernel/scheduler/messaging/SchedulerEventMessageListenerWrapperTest.java (hunk 1 above)
Comment (yuhai) [reply portal-kernel/test/unit/com/liferay/portal/kernel/scheduler/messaging/SchedulerEventMessageListenerWrapperTest.java:243]: Hi @tinatian 

When exposed the bug in the first commit, I wanted to express the receive() executed and then Assert.assertNull(_testMessage2.getResponse()) won't pass. In addition, I thought SchedulerEventMessageListenerWrapper had controlled the listener is not invoked concurrently. So did the change was for test using.

Thanks,
Hai

**Pull request closed without merging**